### PR TITLE
Render config editor, validate & save config to firebase

### PR DESCRIPTION
**Pull request by elayabharath**
+ Added code mirror to display & edit the config json
+ Validate model checks for the missing fields that are required in the config

+ Firebase fetch config returns the raw config instead of converting to model. This change helps me to display the config directly in the code mirror instead of raw -> model -> raw + removing the artefacts of model conversion like enum names. 

+ Reset UI data states when switching back from settings to analyse tab
+ Removed unused methods

### CODE REVIEW

File: lib/firebase.dart [hunk 1]
```diff
@@ -76,13 +75,17 @@ void _fbAuthChanged(
...
   var configSnapshot = await chartsConfigRef.get();
-  var configData = configSnapshot.data();
+  return configSnapshot.data();
+}
 
-  return model.Config.fromData(configData);
+Future<Null> updateConfig(Map<String, dynamic> data) async {
+  var chartsConfigRef = firebase.firestore().doc(fb_constants.metadataPath);
+  var configSnap = await chartsConfigRef.update(data: data);
+  return configSnap;
```
Comment (marianamarasoiu) [inline lib/firebase.dart:88]: We might need to look at catching errors in the places where we talk to firebase, and display them in the UI. Doesn't happen often, and primarily on bad networks, so it's not urgent, just something to add to the todolist :)

File: lib/view.dart [hunk 2]
```diff
@@ -531,14 +535,46 @@ html.SelectElement _getDropdown(String id, List<String> options,
...
-  content.append(html.DivElement()..innerText = 'Settings');
+void renderSettingsTab(String config) {
+  var wrapper = html.DivElement();
+  content.append(wrapper);
+
+  var textArea = html.TextAreaElement()..text = config;
+  wrapper.append(textArea);
+
+  var editor = code_mirror.CodeMirror.fromTextArea(textArea,
+      options: {'mode': 'javascript', 'lineNumbers': true});
```
Comment (marianamarasoiu) [inline lib/view.dart]: does it have a json mode, or is the javascript mode able to suitably display json?

**Review by marianamarasoiu — APPROVED** [review]
LGTM, thanks @elayabharath!

File: lib/view.dart (hunk 2 above)
Comment (elayabharath) [reply lib/view.dart]: Ah yes, javascript is able to display json as it is a subset, but now added explicit mode restricting it to just json.

File: lib/firebase.dart (hunk 1 above)
Comment (elayabharath) [reply lib/firebase.dart:88]: Added a try catch for now to show an alert, but will go through all the firebase functions to make sure they show the errors. Tracking in here: #19

**Comment by elayabharath** [comment]
Thanks @marianamarasoiu

**Pull request merged**